filament/backend/src/vulkan/VulkanDriver.cpp

```diff
@@ -1277,10 +1281,14 @@ void VulkanDriver::beginRenderPass(Handle<HwRenderTarget> rth, const RenderPassP
         .discardEnd = discardEndVal,
         .samples = rt->getSamples(),
         .subpassMask = uint8_t(params.subpassMask),
+        .viewCount = rt->getLayerCount(),
     };
+
     for (int i = 0; i < MRT::MAX_SUPPORTED_RENDER_TARGET_COUNT; i++) {
         const VulkanAttachment& info = rt->getColor(i);
+
         if (info.texture) {
+            rpkey.viewCount = info.texture->getLayerCount();
```

Comment:
This is needed because there are times when the renderTarget's layer count is 2. But the underlying vulkan texture's layer count is 1. If there is a vulkan texture, then the viewcount needs to be set to the value from the texture. Without it validation layers errors occur:

VULKAN ERROR: (Validation) Validation Error: [ VUID-VkFramebufferCreateInfo-renderPass-04536 ] | MessageID = 0x5e2eb360 | vkCreateFramebuffer(): pCreateInfo->pAttachments[1] has a layer count (1) less than or equal to the highest bit in the view mask (1) of subpass 0. The Vulkan spec states: If renderPass was specified with non-zero view masks, each element of pAttachments that is used as an input, color, resolve, or depth/stencil attachment by renderPass must have a layerCount greater than the index of the most significant bit set in any of those view masks (https://vulkan.lunarg.com/doc/view/1.3.268.0/linux/1.3-extensions/vkspec.html#VUID-VkFramebufferCreateInfo-renderPass-04536)

and a segmentation fault crashes the program.

One caveat: If there were ever a case when there is a texture of samplerType::SAMPLER_2D_ARRAY that is not being used in a multiview configuration, it would trigger the multiview structure being added to the renderpass. To get around that. I could pass the value of mContext.isMultiviewEnabled() as part of the RenderPassKey to add an additional check before chaining multiviewCreateInfo to the renderpass.

Reply:
the texture should not have a layer count. But, for now, I propose both RenderTarget and Attachment will both have layerCounts.  You can do an assert here to verify that the render target and attachment counts are the same.

Reply:
Sorry, not vulkan texture, but underlying vulkan image in the texture. When the gtlf_viewer application is not in stereo mode, the layercount of the underlying vulkan image is 1 but the layercount for the vulkan attachment and vulkan rendertarget are both 2. All three (vulkan texture's vkimage, vulkan attachment, and vulkan rendertarget) are all 2, only when the application is in stereo mode. 

Without the current use of the layercount from the vulkan texture's vkImage, the validation error listed above is raised and a crash occurs

Reply:
I think there might be a bug here.  If the rendertarget has 2 layers, then the backing image of the attachments should also have at least 2 layers.  If you see the rendertarget somehow has greater number of layers than any of the attachments, then it's a bug IIUC.

Reply:
Yes, bug fix is in this pr: https://github.com/google/filament/pull/7987